lib/msal-common/src/authority/Authority.ts

```diff
@@ -856,4 +856,13 @@ export class Authority {
 
         return ciamAuthority;
     }
+
+    /**
+     * Extract tenantId from authority
+     */
+    static getTenantFromAuthorityString(authority: string): string | undefined {
+        const authorityUrl = new UrlString(authority);
+        const authorityUrlComponents = authorityUrl.getUrlComponents();
+        return authorityUrlComponents.PathSegments[0];
```

Comment:
This doesn't hold for non-AAD authorities (e.g. B2C) - is this only supported for AAD? Would it perhaps be safer to take the entire path?

Reply:
Fixed by using PathSegments.join("/") so tenant IDs for B2C are covered as well. Will add tests for this specifically as well.